src/pages/ReimbursementAccount/CompanyStep.js

```diff
@@ -46,138 +46,104 @@ class CompanyStep extends React.Component {
+        }
 
-        if (!ValidationUtils.isValidAddress(this.state.addressStreet)) {
-            errors.addressStreet = true;
+        if (!values.addressCity) {
+            errorTexts.addressCity = this.props.translate('bankAccount.error.addressCity');
         }
 
-        if (!ValidationUtils.isValidZipCode(this.state.addressZipCode)) {
-            errors.addressZipCode = true;
+        if (!values.addressState) {
+            errorTexts.addressState = this.props.translate('bankAccount.error.addressState');
         }
 
-        if (!ValidationUtils.isValidURL(this.state.website)) {
-            errors.website = true;
+        if (!values.website || !ValidationUtils.isValidURL(values.website)) {
+            errorTexts.website = this.props.translate('bankAccount.error.website');
         }
 
-        if (!ValidationUtils.isValidTaxID(this.state.companyTaxID)) {
-            errors.companyTaxID = true;
+        if (!values.companyTaxID || !ValidationUtils.isValidTaxID(values.companyTaxID)) {
+            errorTexts.companyTaxID = this.props.translate('bankAccount.error.taxID');
         }
 
-        if (!ValidationUtils.isValidPastDate(this.state.incorporationDate)) {
-            errors.incorporationDate = true;
+        if (!values.incorporationType) {
+            errorTexts.incorporationType = this.props.translate('bankAccount.error.companyType');
         }
 
-        if (!ValidationUtils.isValidUSPhone(this.state.companyPhone, true)) {
-            errors.companyPhone = true;
+        if (!values.incorporationDate || !ValidationUtils.isValidDate(values.incorporationDate)) {
+            errorTexts.incorporationDate = this.props.translate('bankAccount.error.incorporationDate');
         }
 
-        ReimbursementAccount.setBankAccountFormValidationErrors(errors);
+        if (!values.incorporationDate || !ValidationUtils.isValidPastDate(values.incorporationDate)) {
+            errorTexts.incorporationDateFuture = this.props.translate('bankAccount.error.incorporationDateFuture');
+        }
 
-        return _.size(errors) === 0;
-    }
+        if (!values.incorporationState) {
+            errorTexts.incorporationState = this.props.translate('bankAccount.error.incorporationState');
+        }
 
-    submit() {
-        if (!this.validate()) {
-            return;
+        if (!values.companyPhone || !ValidationUtils.isValidUSPhone(values.companyPhone, true)) {
+            errorTexts.companyPhone = this.props.translate('bankAccount.error.phoneNumber');
         }
 
-        const bankAccount = {
-            // Fields from BankAccount step
-            ...ReimbursementAccountUtils.getBankAccountFields(this.props, ['bankAccountID', 'routingNumber', 'accountNumber', 'bankName', 'plaidAccountID', 'plaidAccessToken', 'isSavings']),
+        if (!values.companyName) {
+            errorTexts.companyName = this.props.translate('bankAccount.error.companyName');
+        }
```

Comment:
Let's move this to the top to keep it consistent with the order shown in the form.

Reply:
done